hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/PluralAttributeMappingImpl.java

```diff
@@ -269,6 +271,7 @@ public void finishInitialization(
 						indexFkDescriptor = createForeignKeyDescriptor(
 								( (IndexedCollection) bootDescriptor ).getIndex(),
 								(EntityType) collectionDescriptor.getIndexType(),
+								bootDescriptor.getCollectionTable(),
 								creationProcess,
```

Comment:
The {element} foreign key descriptor created in case of embedded identifier case has bug to have wrong key table. I have to expose a new parameter to pass key table explicitly to **MappingModelCreationHelper .buildEmbeddableForeignKeyDescriptor()** method.